pkg/operator/ceph/cluster/cluster.go

```diff
@@ -614,6 +614,14 @@ func (c *cluster) configureMsgr2() error {
 	}
 	monStore := config.GetMonStore(c.context, c.ClusterInfo)
 
+	encryptionConfig := []config.Option{}
+	for k := range encryptionGlobalConfigSettings {
+		encryptionConfig = append(encryptionConfig, config.Option{Who: "global", Option: k})
+	}
+	if err := monStore.DeleteAll(encryptionConfig...); err != nil {
```

Comment:
With this approach, with every reconcile we will effectively disable encryption before reenabling it. If there is some error condition, perhaps it would be disabled for some time before the operator could reconcile again. How about we only delete the config if we detect that there is a change in the settings? Or if we see that assimilate-conf has settings in the output file that weren't applied, then we delete them and try again. Perhaps the implementation could be inside the `monStore.SetAll()` method.